enda/estimators.py

```diff
@@ -273,103 +339,145 @@ def train(self, df: pandas.DataFrame, target_col: str):
-            prediction_with = prediction_with.reindex(df.index)  # adds rows with NaN values
+            prediction_with = prediction_with.reindex(
+                df.index
+            )  # adds rows with NaN values
 
         if (prediction_with.index != df.index).any():
-            raise ValueError("prediction_with must have the same index as given df. "
-                             "Check that self.estimator_with.predict conserves index.")
+            raise ValueError(
+                "prediction_with must have the same index as given df. "
+                "Check that self.estimator_with.predict conserves index."
+            )
 
         df_without = df.drop(columns=[self.resilient_column])
         prediction_without = self.estimator_without.predict(df_without, target_col)
 
         if (prediction_without.index != df.index).any():
-            raise ValueError("prediction_without must have the same index as given df. "
-                             "Check that self.estimator_without.predict conserves index.")
+            raise ValueError(
+                "prediction_without must have the same index as given df. "
+                "Check that self.estimator_without.predict conserves index."
+            )
 
         return prediction_with, prediction_without
 
     def predict(self, df: pandas.DataFrame, target_col: str) -> pandas.DataFrame:
         predict_with, predict_without = self.predict_both(df, target_col)
 
         # keep prediction with column_name when available, else take the prediction of the estimator without it
-        result = predict_with[target_col].fillna(predict_without[target_col])  # pandas series
+        result = predict_with[target_col].fillna(
+            predict_without[target_col]
+        )  # pandas series
         result = result.to_frame(target_col)
         return result
 
+    @staticmethod
+    def _get_model_name_and_params(model):
+        name = model.model.__class__.__name__
+        params = model.model.get_params()
+        return name, params
+
+    def get_model_params(self):
+        d = {}
+
+        d["estimator_with"] = {"base_estimators": {}, "final_estimators": {}}
+        d["estimator_without"] = {"base_estimators": {}, "final_estimators": {}}
+
+        for _, v in self.estimator_with.base_estimators.items():
+            n, p = self._get_model_name_and_params(v)
+            d["estimator_with"]["base_estimators"][n] = p
+
+        for _, v in self.estimator_without.base_estimators.items():
+            n, p = self._get_model_name_and_params(v)
+            d["estimator_without"]["base_estimators"][n] = p
+
+        n, p = self._get_model_name_and_params(self.estimator_with.final_estimator)
+        d["estimator_with"]["final_estimators"][n] = p
+
+        n, p = self._get_model_name_and_params(self.estimator_without.final_estimator)
+        d["estimator_without"]["final_estimators"][n] = p
+
+        return d
```

Comment:
This shoulf probably outside this class, as it seems it's been built with an `EndaStackedEstimator` in mind, while we're in `EndaEstimatorWithFallback`

Reply:
It's something ad-hoc I added to the class I was using in our own deployment. If anything, it could be added to a common interface between all Enda estimators. We can refactor it in a another PR if you feel like it.

Reply:
Indeed, I think we might work out something to avoid runtime issues (if this function is used with other classes than EndaStackedEstimator).